src/packages/server/projects/control/util.ts

```diff
@@ -145,6 +145,15 @@ export async function createUser(project_id: string): Promise<void> {
   );
 }
 
+
+export async function stopProjectProcesses(project_id: string): Promise<void> {
+    const home_path = homePath(project_id);
+    const pid = await getProjectPID(home_path);
+    const scmd = `kill -9 ${pid} | true `; // | true since kill can return nonzero status. 
+    await exec(scmd); // May be here we should implement some king of GC for processes?
```

Comment:
Yes, my original function deleteProject first does `pkill -9 -u ${uid}`...

Reply:
> king --> kind

Fixed, sorry.


>Also, is there a way to just kill all processes owned by a given uid? 
  
Just killing all processes for the user is not the solution — user can run long running tasks or services, 
runned using tmux/rdp/vnc/etc, completely bypassing cocalc, etc. 

By GC I was thinking about long running zombie processes, descendants of `project.ts` … But still not sure how to reliable select them.

Reply:
> Just killing all processes for the user is not the solution — user can run long running tasks or services,
runned using tmux/rdp/vnc/etc, completely bypassing cocalc, etc. The Idea of this PR is for that.

Ah.  I always think of a stopped project as one where nothing is left running.  However, you do have a good point, and I agree that it would be valuable to have a way of stopping without killing everything.

There is basically one way to stop a project which is to click the "Stop project..." button, and that button has a confirmation popup that says "Stopping the project server will kill all processes....", which is no longer true with this PR in cocalc-docker, but would still be true on cocalc.com. 

<img width="572" alt="image" src="https://user-images.githubusercontent.com/1276278/221688986-e8aa1b7e-7886-4a01-bd18-37ef9c941e80.png">

I'm sorry that this is suddenly getting more complicated, but it is.  

I would happy with this PR if you change it to kill *all* processes. That's the title "Let's don't remove user on stop project" after all.  You could have a follow up PR that introduces a new way to stop a project without killing all processes, but only for cocalc-docker (it doesn't make sense on Kubernetes).  Alternatively, there could be env variable that supports the other behavior for specific cocalc-docker installs.

Reply:
> I would happy with this PR if you change it to kill all processes. 

Okay :(     
Lets merge it.

May be later someone add project settings or ENV variables to disable killing all processes  (for example, integration with Nextcloud, or ...).

Reply:
It is OK now?